apps/prairielearn/src/pages/instructorCourseAdminStaff/instructorCourseAdminStaff.html.ts

```diff
@@ -0,0 +1,748 @@
+import { z } from 'zod';
+
+import { escapeHtml, html } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+
+import { compiledScriptTag } from '../../lib/assets.js';
+import {
+  CourseInstance,
+  CourseInstanceSchema,
+  CourseInstancePermissionSchema,
+  CoursePermissionSchema,
+  User,
+  UserSchema,
+} from '../../lib/db-types.js';
+
+export const CourseUsersRowSchema = z.object({
+  user: UserSchema,
+  course_permission: CoursePermissionSchema,
+  course_instance_roles: z
+    .array(
+      z.object({
+        id: CourseInstanceSchema.shape.id,
+        short_name: CourseInstanceSchema.shape.short_name,
+        course_instance_permission_id: CourseInstancePermissionSchema.shape.id,
+        course_instance_role: CourseInstancePermissionSchema.shape.course_instance_role,
+        course_instance_role_formatted: z.string(),
+      }),
+    )
+    .nullable(),
+  other_course_instances: z
+    .array(
+      z.object({
+        id: CourseInstanceSchema.shape.id,
+        short_name: CourseInstanceSchema.shape.short_name,
+      }),
+    )
+    .nullable(),
+});
+type CourseUsersRow = z.infer<typeof CourseUsersRowSchema>;
+
+let hasUnknownUsers = false;
```

Comment:
It is basically never safe to put state like this at the module level. Remember that this code is going to be running for lots and lots of requests. If you need to get a piece of state into another function, just pass it as an argument!

Reply:
Updated this to a function. Thanks for the help on using `.some()`!